### PR TITLE
Fix completion e2e tests on Windows

### PR DESCRIPTION
**Pull request by cspotcode**
These are the changes we discussed on Discord.

If necessary, I'll be happy to describe them here for posterity; I'm assuming our Discord conversation is sufficient documentation for the moment.

Link to Discord thread:
https://discord.com/channels/226791405589233664/932022470943326339

### CODE REVIEW

File: e2e/shells/bash.test.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ testPty({
   complete: async (bash, request) => {
     const completions = (await bash.write(`\t\t`))
       .join(`\n`)
-      .replaceAll(request.trim(), ``);
+      .replaceAll(`>${request.trim()}`, ``);
```
Comment (paul-soporan) [inline e2e/shells/bash.test.ts]: Should probably be `trimRight` if prefixed with `>`.

```suggestion
      .replaceAll(`>${request.trimRight()}`, ``);
```

File: e2e/shells/pwsh.test.ts [hunk 2]
```diff
@@ -21,12 +21,31 @@ testPty({
...
   complete: async (pwsh, request) => {
-    const output = await pwsh.write(`\t`);
-
-    const completions = output
-      .slice(0, output.findIndex(value => value.includes(request.trim())))
-      .join(`\n`);
+    const output = (await pwsh.write(`\t`)).join(`\n`);
+    console.log({pwshCompleteOutput: output});
+    /*
+     * pwsh MenuCompletion emit varies slightly between posix and windows.
```
Comment (paul-soporan) [inline e2e/shells/pwsh.test.ts]:
```suggestion
     * pwsh MenuComplete emit varies slightly between posix and windows.
```

File: e2e/shells/bash.test.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ chai.use(chaiAsPromised);
 // https://wiki.archlinux.org/title/Bash/Prompt_customization#Prompts
 export const prompts = {
   PS0: ``,
-  PS1: ``,
+  PS1: `>`,
```
Comment (paul-soporan) [inline e2e/shells/bash.test.ts:13]: Can you add a comment here about the empty `PS1` issue?

File: e2e/shells/pwsh.test.ts [hunk 4]
```diff
@@ -21,12 +21,31 @@ testPty({
...
+     * For example, on Windows, we receive this (ANSI escapes omitted, whitespace changed):
+     *   testbin foo --number 3
+     *   3 1 4 2
+     * On Posix we receive this (ANSI escapes omitted, whitespace changed):
+     *   3 1 4 2
+     *   testbin foo --number 3
+     *   3
+     */
+    const requestStart = output.indexOf(request.trim());
+    const requestEnd = output.indexOf(`\r\n`, requestStart) + 2;
```
Comment (paul-soporan) [inline e2e/shells/pwsh.test.ts]: I'd rather normalize the line endings to LF in the `clean` function of `utils.ts`.

File: e2e/shells/bash.test.ts [hunk 5]
```diff
@@ -6,9 +6,11 @@ import {testPty}      from './utils';
 chai.use(chaiAsPromised);
 
 // https://wiki.archlinux.org/title/Bash/Prompt_customization#Prompts
+// For an unknown reason, setting PS1 to an empty string was suppressing
+// completion output when running tests locally.
```
Comment (paul-soporan) [inline e2e/shells/bash.test.ts]:
```suggestion
// completion output when running tests locally on Windows.
```

File: e2e/shells/utils.ts [hunk 6]
```diff
@@ -40,6 +40,7 @@ const clean = (chunk: string) =>
   debug(
     stripAnsi(chunk)
       .replaceAll(BEL, ``)
+      .replaceAll(/\r\n|\r|\n/, `\n`)
```
Comment (paul-soporan) [inline e2e/shells/utils.ts]: `replaceAll` needs to be called with a global regexp. Also, are there any cases on Windows where only `\r` is present without `\n`? Usually when normalizing line endings in Yarn we use `/\r?\n/g`.

```suggestion
      .replaceAll(/\r\n|\r|\n/g, `\n`)
```

Comment (cspotcode) [reply e2e/shells/utils.ts]: I'm not sure to be honest, I wasn't sure if there were cases when dealing with a terminal where you might get \r without \n.

**Pull request merged**